ports/stm32/usb.c

```diff
@@ -579,10 +580,13 @@ STATIC mp_uint_t pyb_usb_vcp_read(mp_obj_t self_in, void *buf, mp_uint_t size, i
 
 STATIC mp_uint_t pyb_usb_vcp_write(mp_obj_t self_in, const void *buf, mp_uint_t size, int *errcode) {
     pyb_usb_vcp_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    if (self->cdc_itf->attached_to_repl) {
+    if ((self->cdc_itf->attached_to_repl) &&
+            (! self->cdc_itf->flow && USBD_CDC_FLOWCONTROL_CTS)) {
```

Comment:
How about removing the `self->cdc_itf->attached_to_repl` check and instead clear CTS in `usb_vcp_attach_to_repl()`?  That gives the user the option to enable/disable CTS even for REPL-attached VCP.

Also: should be `!(self->cdc_itf->flow & USBD_CDC_FLOWCONTROL_CTS)` (note single ampersand).